src/components/UpgradePlanForm/UpgradePlanForm.js

```diff
@@ -4,45 +4,68 @@ import { Formik, Form, Field, ErrorMessage } from 'formik';
 import { InjectAppServices } from '../../services/pure-di';
 import { Loading } from '../../components/Loading/Loading';
 import { SubmitButton } from '../form-helpers/form-helpers';
+import useTimeout from '../../hooks/useTimeout';
 
 const fieldNames = {
   selectedPlanId: 'selectedPlanId',
   message: 'message',
 };
 /** @type { import('../../services/doppler-legacy-client').DopplerLegacyClient } */
-const UpgradePlanForm = ({ handleClose, isSubscriber, dependencies: { dopplerLegacyClient } }) => {
+const UpgradePlanForm = ({
+  handleClose,
+  isSubscriber,
+  dependencies: { dopplerLegacyClient, appSessionRef },
+}) => {
   const [state, setState] = useState({
     availablePlans: null,
     formIsValid: false,
     isLoading: true,
   });
   const intl = useIntl();
+  const createTimeout = useTimeout();
 
   useEffect(() => {
     const fetchData = async () => {
       const availablePlans = await dopplerLegacyClient.getUpgradePlanData(isSubscriber);
+      const filterPlans = availablePlans.filter((p) =>
```

Comment:
La expresión está dificil de leer.  Pudiese usarse desestructuración de objetos. 
```suggestion
const { isSubscribers, maxSubscribers } = appSessionRef.current.userData.user.plan;
const filterPlans = availablePlans.filter((p) => isSubscribers ? p.SubscribersQty > maxSubscribers : p.EmailQty > maxSubscribers);
```

Reply:
Instead of `filterPlans` it should be `availablePlansInSameType` or something other descriptive text.

Reply:
Done, I have a problem with` const { plan } = appSessionRef.current.userData.user;` is not working but witout `{ }` is working 
`const plan  = appSessionRef.current.userData.user;`